src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -290,26 +291,33 @@ public async Task<LSP.CodeAction> HandleRequestAsync(LSP.CodeAction codeAction,
                     Contract.ThrowIfNull(oldTextDoc);
                     Contract.ThrowIfNull(newTextDoc);
 
-                    var oldText = await oldTextDoc.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                    // For linked documents, only generated the document edit once.
+                    if (modifiedDocumentIds.Add(docId))
```

Comment:
This seems fine, as long as something has called https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/Solution.cs,1685 for the changed solution.  That API ensures that changes to the linked files are merged so that both linked documents have exactly the same new text.

We use this in rename for example - https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Rename/RenameHandler.cs,76

This isn't necessarily the case if that is not called (for example if the linked files have #if directives with different changes for each tfm).

Reply:
Sure let me add that

Reply:
OK done